### PR TITLE
Trusted Intermediary: Lab Order Base

### PR DESCRIPTION
**Pull request by halprin**
This PR adds the Trusted Intermediary sender FHIR -> HL7v2 mapping for a Lab Order.  Right now it doesn't specify anything new to the underlying OML mapping, but will in future PRs.

Test Steps:
1. Ran a [Trusted Intermediary example lab order](https://github.com/CDCgov/trusted-intermediary/blob/775954caa49fb44b8830c53e27c95cc63e73670a/examples/fhir/lab_order.json) through ReportStream.  `./gradlew primeCLI --args="fhirdata -i ../../trusted-intermediary/examples/fhir/lab_order.json --output-format HL7 -s metadata/hl7_mapping/Senders/TrustedIntermediaryLabOrder.yml"`

## Changes
- Added `TrustedIntermediaryLabOrder.yml` which will act as the basis for our sender template.
- Modified `message-date-time` to get the date and time stamp from another possible location.  This is according to this [official FHIR/HL7v2 spreadsheet](https://docs.google.com/spreadsheets/d/1oOy6JW4Mv5VL8DsoA05bYZCUa1JqP4GCYNPF5GJTLxQ/edit#gid=0).
  - In doing this, I converted it to use YAML array format instead of JSON array format  (e.g. the use of dashes `-` instead of brackets `[` `]`).  Would we rather be consistent?

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [x] Added tests?  _None needed, I believe_.

### Process
- [x] Are there licensing issues with any new dependencies introduced?  _No_.
- [x] Includes a summary of what a code reviewer should test/verify?
- [x] Updated the release notes?  _Not needed, I believe_.
- [x] Database changes are submitted as a separate PR?  _No database changes_.
- [x] DevOps team has been notified if PR requires ops support?  _No ops support_.

## Linked Issues
- Helps towards https://github.com/CDCgov/trusted-intermediary/issues/79

## To Be Done

_N/A_.

## Specific Security-related subjects a reviewer should pay specific attention to

- Does this PR introduce new endpoints?  _No_.
- Does this PR include changes in authentication and/or authorization of existing endpoints?  _No_.
- Does this change introduce new dependencies that need vetting?  _No new dependencies_.
- Does this change require changes to our infrastructure?  _No_.
- Does logging contain sensitive data?  _No logging affected_.
- Does this PR include or remove any sensitive information itself?  _No_.

### CODE REVIEW

File: prime-router/metadata/hl7_mapping/ORU_R01/base/message-header.yml [hunk 1]
```diff
@@ -84,7 +84,9 @@ elements:
     hl7Spec: [ MSH-6-3 ]
 
   - name: message-date-time
-    value: [ 'Bundle.entry.resource.ofType(Provenance).recorded' ]
+    value:
+      - Bundle.entry.resource.ofType(Provenance).recorded
+      - Bundle.timestamp
```
Comment (arnejduranovic) [inline prime-router/metadata/hl7_mapping/ORU_R01/base/message-header.yml:89]: I spoke to @carlosfelix2 regarding this change. `Bundle.timestamp` does not correctly store the timezone data, so `Bundle.entry.resource.ofType(Provenance).recorded` is preferred but `Bundle.timestamp` makes a good backup option if the preferred data is not so this is ultimately a good change.

File: prime-router/metadata/hl7_mapping/Senders/TrustedIntermediaryLabOrder.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+extends: ../OML_O21/OML_O21-base
```
Comment (arnejduranovic) [inline prime-router/metadata/hl7_mapping/Senders/TrustedIntermediaryLabOrder.yml:1]: This folder set-up is temporary for now. Sender transforms are still being worked on and once that is done the place where this file exists will change, but its fine here for now.

**Review by arnejduranovic — APPROVED** [review]
Great job! Reviewed this with @carlosfelix2 and looks good to merge.

File: prime-router/metadata/hl7_mapping/Senders/TrustedIntermediaryLabOrder.yml (hunk 2 above)
Comment (halprin) [reply prime-router/metadata/hl7_mapping/Senders/TrustedIntermediaryLabOrder.yml:1]: Speaking of "sender transforms" which are still a work in progress, I see our `prime-router/settings/staging/0149-etor.yml` which mentions the `OML_O21-base` YAML file (line `58`) for a _receiver_.  Should that be updated to the new `TrustedIntermediaryLabOrder.yml` file that is in this PR?  Line `58` is under receiving though, and this `TrustedIntermediaryLabOrder.yml` file is all about sending.  So maybe we shouldn't update line `58` to the new file.  Thoughts?

Comment (arnejduranovic) [reply prime-router/metadata/hl7_mapping/Senders/TrustedIntermediaryLabOrder.yml:1]: I'd say let's wait and see how all this shakes out once everything is restructured, then we can think about how all this is organized. We can investigate issues as they come up and see if we need to update these files.

Comment (halprin) [reply prime-router/metadata/hl7_mapping/Senders/TrustedIntermediaryLabOrder.yml:1]: Sounds like a plan.

**Pull request merged**